<?xml version="1.0" ?>
<launch>
  <arg name="visualize" default="true" />
  <arg name="sensor" default="realsense"/>

  <!-- Object detection -->
  <node pkg="fetch_demo" type="object_detection_node.py" name="object_detection_node" output="screen"/>

  <node name="mask_rcnn" pkg="mask_rcnn_ros" type="mask_rcnn_node.py" output="log">
    <remap from="~input" to="/camera/color/image_raw" />
    <param name="~visualization" value="$(arg visualize)" />
  </node>

  <include file="$(find depth_segmentation)/launch/depth_segmentation.launch">
    <arg name="depth_segmentation_params_file" value="$(find depth_segmentation)/cfg/$(arg sensor)_config.yaml"/>
    <arg name="sensor_topics_file" value="$(find depth_segmentation)/cfg/$(arg sensor)_topics.yaml"/>
    <param name="semantic_instance_segmentation/enable" value="true"/>
    <param name="label_display" value="$(arg visualize)"/>
  </include>

  <arg name="gsm_node_params_name" default="$(find gsm_node)/cfg/yumi.yaml"/>

  <node name="gsm_node" pkg="gsm_node" type="gsm_node" output="screen">
    <rosparam command="load" file="$(arg gsm_node_params_name)" />
    <param name="meshing/visualize" value="$(arg visualize)"/>
    <param name="meshing/update_mesh_every_n_sec" value="0.0" unless="$(arg visualize)"/>
  </node>
</launch>
